### PR TITLE
Integrated form submission functionality

### PR DESCRIPTION
**Pull request by Polarts**

### CODE REVIEW

File: src/components/ProjectRequestForm/FormConfirmation.jsx [hunk 1]
```diff
@@ -155,18 +159,25 @@ export default function FormConfirmation({
...
-      }
+      <div className={styles.bottomContainer}>
+        {
+          !response.result // The result is false
+          ? <Alert variant="danger">{response.error}</Alert> 
+          : null
+        }
+        { 
+          isBusy
+          ? <LoadingDots/>
```
Comment (Tomer51m) [inline src/components/ProjectRequestForm/FormConfirmation.jsx:170]: Please change the color of the dot from black to the website green.

Comment (Tomer51m) [inline src/components/ProjectRequestForm/FormConfirmation.jsx:170]: Please change the dots color from black to the website's green.

File: src/services/requestForm.service.js [hunk 2]
```diff
@@ -1,16 +1,46 @@
...
+        if (!formBody.webAddress.length) {
+            // the server won't approve empty string.
+            delete formBody.webAddress;
+        }
+
+        let response = await fetch(`${URL}/${isForProfit? 'profitableProjectReqs' : 'charitableProjectReqs'}`, {
             method: "POST",
-            headers: {
+            body: JSON.stringify(formBody),
+            headers:{
```
Comment (Tomer51m) [inline src/services/requestForm.service.js:30]: The convention is that headers should be before the body. Make the code more readable.

File: src/services/requestForm.service.js [hunk 3]
```diff
@@ -1,16 +1,46 @@
...
+    email: "How can we contact you without an email?",
+    phone: "How can we contact you without a phone number?",
+    about: "You forgot to provide details about your project!",
+    description: "You forgot to provide details about your organization!",
+    webAddress: "You should never see this error message. Like, never.",
+    tasks: "You haven't told us what needs to be done!"
+};
+
+function mapErrorMessage(errorMsg) {
+    if (errorMsg.startsWith('"')) {
```
Comment (Tomer51m) [inline src/services/requestForm.service.js:14]: Not very robust. Would be better to match expected response from server.
Also, you forgot to handle 'internal server error' message that the server can respond with.

**Review by Tomer51m — CHANGES_REQUESTED** [review]

**Comment by Polarts** [comment]
@Tomer51m I addressed your comments, and also added a feature to mark the error row in the table with alert-danger. Please take a look

**Review by Tomer51m — APPROVED** [review]

**Pull request merged**